### PR TITLE
Update Steam documentation for config flow

### PR DESCRIPTION
**Pull request by tkdrob**
## Proposed change

Update Steam documentation for config flow

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/67261
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/steam_online.markdown [hunk 1]
```diff
@@ -3,46 +3,26 @@ title: Steam
...
 ha_release: 0.14
 ha_domain: steam_online
 ha_platforms:
   - sensor
+ha_codeowners:
+  - '@tkdrob'
 ---
 
-The `steam` sensor platform will allow you to track the online status of public [Steam](https://steamcommunity.com) accounts.
+The `steam` integration will allow you to track the online status of public [Steam](https://steamcommunity.com) accounts.
```
Comment (frenck) [inline source/_integrations/steam_online.markdown]:
```suggestion
The Steam integration will allow you to track the online status of public [Steam](https://steamcommunity.com) accounts.
```

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Comment by KevinCathcart** [comment]
Should be retargeted at rc, since the parent is already merged and included in the beta.

Merging into rc will fix #22574 

Also, reminder: there is post merge review feedback on the parent PR that should be fixed with a subsequent PR.

**Comment by frenck** [comment]
> Should be retargeted at rc, since the parent is already merged and included in the beta.

No, it should not. We don't use `rc` branch for PRs directly.

**Review by frenck — APPROVED** [review]
Thanks, @tkdrob 👍

**Pull request merged**